File: clients/java/src/main/java/komus24/model/Traffic.java
package komus24.model;

import komus24.util.StreamUtil;

/**
 * TODO - Document
 */
public class Traffic {
    /**
     * TODO - Document
     */
    private int amount;

    /**
     * TODO - Document
     */
    public int getAmount() {
        return amount;
    }

    /**
     * TODO - Document
     */
    public void setAmount(int value) {
        this.amount = value;
    }
    /**
     * TODO - Document
     */
    private double moveTime;

    /**
     * TODO - Document
     */
    public double getMoveTime() {
        return moveTime;
    }

    /**
     * TODO - Document
     */
    public void setMoveTime(double value) {
        this.moveTime = value;
    }
    /**
     * TODO - Document
     */
    private double radius;

    /**
     * TODO - Document
     */
    public double getRadius() {
        return radius;
    }

    /**
     * TODO - Document
     */
    public void setRadius(double value) {
        this.radius = value;
    }
    /**
     * TODO - Document
     */
    private double weight;

    /**
     * TODO - Document
     */
    public double getWeight() {
        return weight;
    }

    /**
     * TODO - Document
     */
    public void setWeight(double value) {
        this.weight = value;
    }
    /**
     * TODO - Document
     */
    private double crashDeceleration;

    /**
     * TODO - Document
     */
    public double getCrashDeceleration() {
        return crashDeceleration;
    }

    /**
     * TODO - Document
     */
    public void setCrashDeceleration(double value) {
        this.crashDeceleration = value;
    }
    /**
     * TODO - Document
     */
    private double crashLifetime;

    /**
     * TODO - Document
     */
    public double getCrashLifetime() {
        return crashLifetime;
    }

    /**
     * TODO - Document
     */
    public void setCrashLifetime(double value) {
        this.crashLifetime = value;
    }

    public Traffic(int amount, double moveTime, double radius, double weight, double crashDeceleration, double crashLifetime) {
        this.amount = amount;
        this.moveTime = moveTime;
        this.radius = radius;
        this.weight = weight;
        this.crashDeceleration = crashDeceleration;
        this.crashLifetime = crashLifetime;
    }

    /**
     * Read Traffic from input stream
     */
    public static Traffic readFrom(java.io.InputStream stream) throws java.io.IOException {
        int amount;
        amount = StreamUtil.readInt(stream);
        double moveTime;
        moveTime = StreamUtil.readDouble(stream);
        double radius;
        radius = StreamUtil.readDouble(stream);
        double weight;
        weight = StreamUtil.readDouble(stream);
        double crashDeceleration;
        crashDeceleration = StreamUtil.readDouble(stream);
        double crashLifetime;
        crashLifetime = StreamUtil.readDouble(stream);
        return new Traffic(amount, moveTime, radius, weight, crashDeceleration, crashLifetime);
    }

    /**
     * Write Traffic to output stream
     */
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeInt(stream, amount);
        StreamUtil.writeDouble(stream, moveTime);
        StreamUtil.writeDouble(stream, radius);
        StreamUtil.writeDouble(stream, weight);
        StreamUtil.writeDouble(stream, crashDeceleration);
        StreamUtil.writeDouble(stream, crashLifetime);
    }

    /**
     * Get string representation of Traffic
     */
    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder("Traffic { ");
        stringBuilder.append("amount: ");
        stringBuilder.append(String.valueOf(amount));
        stringBuilder.append(", ");
        stringBuilder.append("moveTime: ");
        stringBuilder.append(String.valueOf(moveTime));
        stringBuilder.append(", ");
        stringBuilder.append("radius: ");
        stringBuilder.append(String.valueOf(radius));
        stringBuilder.append(", ");
        stringBuilder.append("weight: ");
        stringBuilder.append(String.valueOf(weight));
        stringBuilder.append(", ");
        stringBuilder.append("crashDeceleration: ");
        stringBuilder.append(String.valueOf(crashDeceleration));
        stringBuilder.append(", ");
        stringBuilder.append("crashLifetime: ");
        stringBuilder.append(String.valueOf(crashLifetime));
        stringBuilder.append(" }");
        return stringBuilder.toString();
    }
}

File: clients/kotlin/src/main/kotlin/komus24/model/Traffic.kt
package komus24.model

import komus24.util.StreamUtil

/**
 * Options for traffic
 */
class Traffic {
    /**
     * Number of traffic cars
     */
    var amount: Int
    /**
     * Time to move between adjacent keypoints
     */
    var moveTime: Double
    /**
     * Radius of each traffic car
     */
    var radius: Double
    /**
     * Weight of each traffic car
     */
    var weight: Double
    /**
     * Deceleration after crash
     */
    var crashDeceleration: Double
    /**
     * Lifetime after crash
     */
    var crashLifetime: Double

    constructor(amount: Int, moveTime: Double, radius: Double, weight: Double, crashDeceleration: Double, crashLifetime: Double) {
        this.amount = amount
        this.moveTime = moveTime
        this.radius = radius
        this.weight = weight
        this.crashDeceleration = crashDeceleration
        this.crashLifetime = crashLifetime
    }

    /**
     * Write Traffic to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, amount)
        StreamUtil.writeDouble(stream, moveTime)
        StreamUtil.writeDouble(stream, radius)
        StreamUtil.writeDouble(stream, weight)
        StreamUtil.writeDouble(stream, crashDeceleration)
        StreamUtil.writeDouble(stream, crashLifetime)
    }

    /**
     * Get string representation of Traffic
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("Traffic { ")
        stringBuilder.append("amount: ")
        stringBuilder.append(amount)
        stringBuilder.append(", ")
        stringBuilder.append("moveTime: ")
        stringBuilder.append(moveTime)
        stringBuilder.append(", ")
        stringBuilder.append("radius: ")
        stringBuilder.append(radius)
        stringBuilder.append(", ")
        stringBuilder.append("weight: ")
        stringBuilder.append(weight)
        stringBuilder.append(", ")
        stringBuilder.append("crashDeceleration: ")
        stringBuilder.append(crashDeceleration)
        stringBuilder.append(", ")
        stringBuilder.append("crashLifetime: ")
        stringBuilder.append(crashLifetime)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read Traffic from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Traffic {
            var amount: Int
            amount = StreamUtil.readInt(stream)
            var moveTime: Double
            moveTime = StreamUtil.readDouble(stream)
            var radius: Double
            radius = StreamUtil.readDouble(stream)
            var weight: Double
            weight = StreamUtil.readDouble(stream)
            var crashDeceleration: Double
            crashDeceleration = StreamUtil.readDouble(stream)
            var crashLifetime: Double
            crashLifetime = StreamUtil.readDouble(stream)
            return Traffic(amount, moveTime, radius, weight, crashDeceleration, crashLifetime)
        }
    }
}